### PR TITLE
[master] Upgrade evm-ds to latest version 0.37

### PR DESCRIPTION
**Pull request by n-hutton**
Most notably the precompiles want the cost to be returned as well, now

### CODE REVIEW

File: evm-ds/src/convert.rs [hunk 1]
```diff
@@ -178,3 +181,153 @@ impl From<(Bytes, Bytes)> for EvmProto::Storage {
         result
     }
 }
+
+fn as_string(opcode: &Opcode) -> String {
```
Comment (valeryz) [inline evm-ds/src/convert.rs:185]: This should rather be done as `impl fmt::Display for Opcode`

Comment (valeryz) [reply evm-ds/src/convert.rs:185]: which you can as well implement by calling into fmt::Debug imlementation that Rust already provides for enums

File: evm-ds/src/main.rs [hunk 2]
```diff
@@ -3,6 +3,8 @@
 // #![deny(warnings)]
 #![forbid(unsafe_code)]
 
+extern crate core;
```
Comment (valeryz) [inline evm-ds/src/main.rs:6]: I'm not sure why this is needed.

**Review by valeryz — APPROVED** [review]
Thanks, all seems good, but a few suggestions about the Rust style.

File: evm-ds/src/main.rs (hunk 2 above)
Comment (n-hutton) [reply evm-ds/src/main.rs]: It's not

File: evm-ds/src/convert.rs (hunk 1 above)
Comment (valeryz) [reply evm-ds/src/convert.rs:185]: Indeed, as discussed in chat, outside evm crate you can't do it, please ignore the above.

**Pull request merged**